File: Codinsa2015.Client/Codinsa2015.Client/Views/Clank/ClientCpp/src/Vector2.cpp
#include "../inc/Vector2.h"
Vector2::Vector2()
{
}
Vector2::Vector2(float x, float y)
{
	this->X = x;
	this->Y = y;
}
void Vector2::serialize(std::ostream& output) {
	// X
	output << ((float)this->X) << '\n';
	// Y
	output << ((float)this->Y) << '\n';
}

Vector2 Vector2::deserialize(std::istream& input) {
	Vector2 _obj = Vector2();
	// X
	float _obj_X; input >> _obj_X; input.ignore(1000, '\n');
	_obj.X = (float)_obj_X;
	// Y
	float _obj_Y; input >> _obj_Y; input.ignore(1000, '\n');
	_obj.Y = (float)_obj_Y;
	return _obj;
}

File: Codinsa2015.Client/Codinsa2015.Client/Views/Clank/ClientJava/src/net/codinsa2015/Vector2.java
package net.codinsa2015;
import java.lang.*;
import java.util.ArrayList;
import java.io.BufferedReader;
import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.io.UnsupportedEncodingException;
import net.codinsa2015.Vector2.*;


@SuppressWarnings("unused")
public class Vector2
{


	public Vector2()
	{
	}
	public Vector2(Float x, Float y)
	{
		this.X = x;
		this.Y = y;
	}
	public Float X;
	public Float Y;
	public Vector2() {
	}

	public static Vector2 deserialize(BufferedReader input) throws UnsupportedEncodingException, IOException {
		Vector2 _obj =  new Vector2();
		// X
		float _obj_X = Float.valueOf(input.readLine());
		_obj.X = _obj_X;
		// Y
		float _obj_Y = Float.valueOf(input.readLine());
		_obj.Y = _obj_Y;
		return _obj;
	}

	public void serialize(OutputStreamWriter output) throws UnsupportedEncodingException, IOException {
		// X
		output.append(((Float)this.X).toString() + "\n");
		// Y
		output.append(((Float)this.Y).toString() + "\n");
	}

}
